From 74e1135362573f8db0d28f97e9eee64a3ca86ee6 Mon Sep 17 00:00:00 2001
From: hongruichen <chraac@gmail.com>
Date: Wed, 10 Apr 2024 22:48:14 +0800
Subject: allwinner-common-h616-THS-workaround.patch

---
 board/sunxi/board.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/board/sunxi/board.c b/board/sunxi/board.c
index 1313b01dce..ec7f9bf6d0 100644
--- a/board/sunxi/board.c
+++ b/board/sunxi/board.c
@@ -226,6 +226,15 @@ int board_init(void)
 	if (ret)
 		return ret;
 
+#if CONFIG_MACH_SUN50I_H616
+	/*
+	 * The bit[16] of register reg[0x03000000] must be zero for the THS
+	 * driver to work properly in the kernel. The BSP u-boot is putting
+	 * the whole register to zero so we are doing the same.
+	 */
+	writel(0x0, SUNXI_SRAMC_BASE);
+#endif
+
 	eth_init_board();
 
 	return 0;
-- 
GitLab
